compiler/rustc_codegen_ssa/src/back/metadata.rs

```diff
@@ -273,6 +273,11 @@ pub(crate) fn create_object_file(sess: &Session) -> Option<write::Object<'static
                 e_flags |= elf::EF_RISCV_RVC;
             }
 
+            // Check if embedded is enabled
```

Comment:
```suggestion
            // Specify if the embedded abi is used
```

It seems that the embedded abi can be used even in programs that use the full RV32I instruction set[^1]. This flag only determines if the embedded abi is used as I understand it[^2], not if the object file is limited to RV32E instructions.

[^1]: https://github.com/riscv-non-isa/riscv-eabi-spec/blob/master/EABI.adoc#1-goals
[^2]: https://github.com/riscv-non-isa/riscv-elf-psabi-doc/blob/master/riscv-elf.adoc#file-header